c/src/ml-api-inference-pipeline.c

```diff
@@ -915,7 +917,8 @@ iterate_element (ml_pipeline * pipe_h, GstElement * pipeline,
               if (element_type != ML_PIPELINE_ELEMENT_UNKNOWN) {
                 ml_pipeline_element *e;
 
-                e = construct_element (elem, pipe_h, name, element_type);
+                e = construct_element (gst_object_ref (elem), pipe_h, name,
```

Comment:
let's consider failure case if construct_element() returns null, at line 930 - allocation failure,
```
} else {
  /* allocation failure */
  gst_object_unref (elem);
```

Reply:
Okay, I will add it.